synapse/handlers/federation.py

```diff
@@ -565,14 +566,14 @@ async def do_invite_join(
             except PartialStateConflictError as e:
                 # The homeserver was already in the room and it is no longer partial
                 # stated. We ought to be doing a local join instead. Turn the error into
-                # a 503, as a hint to the client to try again.
+                # a 429, as a hint to the client to try again.
                 # TODO(faster_joins): `_should_perform_remote_join` suggests that we may
                 #   do a remote join for restricted rooms even if we have full state.
                 logger.error(
                     "Room %s was un-partial stated while processing remote join.",
                     room_id,
                 )
-                raise SynapseError(HTTPStatus.SERVICE_UNAVAILABLE, e.msg, e.errcode)
+                raise LimitExceededError(msg=e.msg, errcode=e.errcode, retry_after_ms=0)
```

Comment:
should we consider opening an issue as to talk about whether we want to make this better?
I see what you're saying but it still feels to me that this is worth thinking about (but I don't want to block this PR on that).
(For sending messages, some clients seem to prompt you to *retry* sending the message if it fails, I'm not sure about the exact circumstances but leaving it like this means we'll want to check that, so perhaps defer it to an issue regardless)

Reply:
that's fair. I've filed it as #13173.